hive/src/main/java/com/netflix/iceberg/hive/HiveTables.java

```diff
@@ -53,20 +56,42 @@ public Table create(Schema schema, String tableIdentifier) {
 
   @Override
   public Table create(Schema schema, PartitionSpec spec, Map<String, String> properties, String tableIdentifier) {
-    List<String> parts = DOT.splitToList(tableIdentifier);
-    if (parts.size() == 2) {
-      return create(schema, spec, properties, parts.get(0), parts.get(1));
-    }
-    throw new UnsupportedOperationException("Could not parse table identifier: " + tableIdentifier);
+    final List<String> parts = parseTableIdentifier(tableIdentifier);
+    return create(schema, spec, properties, parts.get(0), parts.get(1));
   }
 
   @Override
   public Table load(String tableIdentifier) {
-    List<String> parts = DOT.splitToList(tableIdentifier);
-    if (parts.size() == 2) {
-      return load(parts.get(0), parts.get(1));
+    final List<String> parts = parseTableIdentifier(tableIdentifier);
+    return load(parts.get(0), parts.get(1));
+  }
+
+  @Override
+  public void drop(String tableIdentifier) {
+    final List<String> parts = parseTableIdentifier(tableIdentifier);
+    try {
+      getClient().drop_table(parts.get(0), parts.get(1), false);
```

Comment:
the ` boolean deleteData` can be bubbled up as a flag to the api so caller can choose to enforce data deletion.